avocado/plugins/run.py

```diff
@@ -166,7 +168,17 @@ def run(self, args):
             log.error(e.message)
             sys.exit(exit_codes.AVOCADO_FAIL)
         job_instance = job.Job(args)
+
+        # Run JobPre plugins
+        pre_post_dispatcher = JobPrePostDispatcher()
+        output.log_plugin_failures(pre_post_dispatcher.load_failures)
+        pre_post_dispatcher.map_method('pre', job_instance)
+
         job_run = job_instance.run()
+
+        # Run JobPost plugins
+        pre_post_dispatcher.map_method('post', job_instance)
```

Comment:
Actually regarding the style, what would you say about splitting the `run` into 3 stages? I know there are some users overriding the `run` plugin, perhaps it would be useful for them to be able to do something {before, after} the {pre, run, post} via inheritance.

Anyway this is just a question as I think those use users should move to `Pre/Post` plugins instead of inheriting the class...